zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -303,6 +437,40 @@ final class ZQuery[-R, +E, +A] private (private val step: ZIO[(R, QueryContext),
   final def timed: ZQuery[R with Clock, E, (Duration, A)] =
     summarized(clock.nanoTime)((start, end) => Duration.fromNanos(end - start))
 
+  /**
+   * Moves a `None` value in the error channel into the value channel while converting the existing value into a `Some`
+   *
+   * Inverse of [[ZQuery.some]]
+   */
+  def unSome[E1](implicit ev: E <:< Option[E1]): ZQuery[R, E1, Option[A]] =
```

Comment:
This one is definitely the weirdest one. The issue being that there is already an `optional` on `ZQuery` which has a different behavior to what is on `ZIO`. Thus I tried to map a common pattern that occurs in `ZStream` (`unTake` I think it was), however I'm not sure if there are better names. Given that ZQuery is not yet in 1.0 one option is to change the semantics of `optional` but I don't know what we would call the new operator that would inherit its current behavior.

Reply:
Yes. I think ZIO Stream ended up going with `collectSome` which is not bad. It would be nice to have consistency with ZIO but not sure what another name for that is that would be logical and to a certain extent I think that is a more "important" operator since it does something to query execution that is relatively primitive while this is just a convenience method. Maybe we go with `collectSome` for now and can change if we come up with a better name for the current `optional`? Definitely open to ideas there!